File: WrappersNET/MathLibNETTest/TestSub.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace MathLibNETTest
{
    [TestClass]
    public class TestSub
    {
        [TestMethod]
        [TestCategory("Sub")]
        [TestCategory("MathLib")]
        public void TestSubtraction1()
        {
            double iVal = MathLibNET.Math.Sub(-1, -2);
            Assert.AreEqual(iVal, 1);
        }

        [TestMethod]
        [TestCategory("Sub")]
        [TestCategory("MathLib")]
        public void TestSubtraction2()
        {
            double iVal = MathLibNET.Math.Sub(-1, -1);
            Assert.AreEqual(iVal, 0);
        }

        [TestMethod]
        [TestCategory("Sub")]
        [TestCategory("MathLib")]
        public void TestSubtraction3()
        {
            double iVal = MathLibNET.Math.Sub(0, 0);
            Assert.AreEqual(iVal, 0);
        }

        [TestMethod]
        [TestCategory("Sub")]
        [TestCategory("MathLib")]
        public void TestSubtraction4()
        {
            double iVal = MathLibNET.Math.Sub(-1, -2);
            Assert.AreEqual(iVal, 1);
        }

        [TestMethod]
        [TestCategory("Sub")]
        [TestCategory("MathLib")]
        public void TestSubtraction5()
        {
            double iVal = MathLibNET.Math.Sub(10, 5);
            Assert.AreEqual(iVal, 5);
        }
    }
}

File: Native/MathLibTest/TestSub.cpp
#include "stdafx.h"
#include "CppUnitTest.h"
#include "MathLib.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace MathLibTest
{
    TEST_CLASS(TestSub)
    {
    public:
        BEGIN_TEST_METHOD_ATTRIBUTE(TestSub1)
            TEST_METHOD_ATTRIBUTE(L"Category", L"MathLib")
            TEST_METHOD_ATTRIBUTE(L"Category", L"Sub")
        END_TEST_METHOD_ATTRIBUTE()
        TEST_METHOD(TestSub1)
        {
            double d = Math::Sub(-5, 10);
            Assert::AreEqual(-15.0, d);
        }

        BEGIN_TEST_METHOD_ATTRIBUTE(TestSub2)
            TEST_METHOD_ATTRIBUTE(L"Category", L"MathLib")
            TEST_METHOD_ATTRIBUTE(L"Category", L"Sub")
        END_TEST_METHOD_ATTRIBUTE()
        TEST_METHOD(TestSub2)
        {
            double d = Math::Sub(0.0, 0.0);
            Assert::AreEqual(0.0, d);
        }

        BEGIN_TEST_METHOD_ATTRIBUTE(TestSub3)
            TEST_METHOD_ATTRIBUTE(L"Category", L"MathLib")
            TEST_METHOD_ATTRIBUTE(L"Category", L"Sub")
        END_TEST_METHOD_ATTRIBUTE()
        TEST_METHOD(TestSub3)
        {
            double d = Math::Sub(-1.0, 1.0);
            Assert::AreEqual(-2.0, d);
        }

        BEGIN_TEST_METHOD_ATTRIBUTE(TestSub4)
            TEST_METHOD_ATTRIBUTE(L"Category", L"MathLib")
            TEST_METHOD_ATTRIBUTE(L"Category", L"Sub")
        END_TEST_METHOD_ATTRIBUTE()
        TEST_METHOD(TestSub4)
        {
            double d = Math::Sub(1.0, -1.0);
            Assert::AreEqual(2.0, d);
        }
    };
}